src/app/route/CategoryPage/CategoryPage.component.js

```diff
@@ -142,79 +164,105 @@ class CategoryPage extends Component {
         return path.indexOf('search') === 0 ? null : path;
     }
 
+    /**
+     * Get search parameter for request
+     * @return {String} search request parameter
+     */
+    getSearchParam() {
+        const search = getQueryParam('search', location);
+        return search ? decodeURIComponent(search) : '';
+    }
+
     /**
      * Check if url was changed
      * @return {Boolean}
      */
     urlHasChanged(location, prevProps) {
         const pathnameHasChanged = location.pathname !== prevProps.location.pathname;
-        const searchQueryHasChanged = location.search !== prevProps.location.search;
+        const searchQueryHasChanged = !this.compareQueriesWithoutPage(location.search, prevProps.location.search);
 
         return pathnameHasChanged || searchQueryHasChanged;
     }
 
+    /**
+     * Compares queries with specified filter passed in callback
+     * @returns {Boolean}
+     */
+    compareQueriesWithFilter(currentQuery, previousQuery, callback) {
+        const currentParams = callback(convertQueryStringToKeyValuePairs(currentQuery));
+        const previousParams = callback(convertQueryStringToKeyValuePairs(previousQuery));
+
+        return JSON.stringify(currentParams) === JSON.stringify(previousParams);
+    }
+
+    /**
+     * Compares queries ignoring page number
+     * @return {Boolean}
+     */
+    compareQueriesWithoutPage(currentQuery, previousQuery) {
+        const removePage = (params) => {
+            const { page, ...filteredParams } = params;
+
+            return filteredParams;
+        };
+
+        return this.compareQueriesWithFilter(currentQuery, previousQuery, removePage);
+    }
+
+    /**
+     * Compares queries ignoring sortKey and sortDirection
+     * @return {Boolean}
+     */
+    compareQueriesWithoutSort(currentQuery, previousQuery) {
+        const removeSortKeyAndDirection = (params) => {
+            const { sortKey, sortDirection, ...filteredParams } = params;
```

Comment:
Unused? Hmm?

Reply:
The function is used on line 390, and it is done to prevent requesting ProductListInfo on sorting change, as it does not bring any new information (like we are just rearranging thinks). But 'excluders' like that is one of the reason why I do not like ProductListInfo as it is now. In my opinion, all filters, sortData, etc should be placed in category query and be requested per category (however it requires reworking category requesting mechanics to be efficient). And totalPages requested in ProductList qurey (so changes of ~someday~ working minmax would refresh totalCount as well).